### PR TITLE
[MB-1174] Prevent node from starting if using 0.0.0.0 as thrift server host

### PR DESCRIPTION
**Pull request by AnujaLK**
JIRA - https://wso2.org/jira/browse/MB-1174

### CODE REVIEW

**Comment by pumudu88** [comment]
can't we do this validation in AndesConfigurationManager?
since we only validating 0.0.0.0 here, can we set localhost address(127.0.0.1) by deafult if some invalid ip is given? For examle we need to validate for * as well.

**Comment by AnujaLK** [comment]
Validating and throwing exception in MBThriftServer instead of AndesConfigurationManager will avoid adding unwanted tasks to configuration manager.

File: modules/andes-core/broker/src/main/java/org/wso2/andes/thrift/MBThriftServer.java [hunk 1]
```diff
@@ -65,8 +65,13 @@ private MBThriftServer() {
     public void start(final String hostName,
                       final int port,
                       final String taskName) throws AndesException {
+        /**
+         * Stop node if 0.0.0.0 used as thrift server host in broker.xml
+         */
+        if ("0.0.0.0".equals(hostName)) {
+            throw new RuntimeException("Invalid thrift server host 0.0.0.0");
```
Comment (ramith) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/thrift/MBThriftServer.java]: why not throw a andesexception?

Comment (AnujaLK) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/thrift/MBThriftServer.java]: Fixed with PR https://github.com/wso2/andes/pull/276

**Pull request merged**